### PR TITLE
MAYA-122131 Maya Ref Auto-Edit Handling

### PR DESCRIPTION
**Pull request by pierrebai-adsk**
Turn off auto-edit when ref edit are discarded.

### CODE REVIEW

File: lib/mayaUsd/CMakeLists.txt [hunk 1]
```diff
@@ -200,6 +200,7 @@ target_link_libraries(${PROJECT_NAME}
         $<$<BOOL:${UFE_FOUND}>:${UFE_LIBRARY}>
         ${MAYA_LIBRARIES}
         mayaUsdUtils
+        mayaUsd_Schemas
```
Comment (ppt-adsk) [inline lib/mayaUsd/CMakeLists.txt:203]: Don't think we should be adding this dependence.

File: lib/mayaUsd/fileio/primUpdaterManager.cpp [hunk 2]
```diff
@@ -1022,6 +1023,15 @@ bool PrimUpdaterManager::discardPrimEdits(const Ufe::Path& pulledPath)
         updater->discardEdits();
     }
 
+    // Reset the auto-edit when discarding the edit.
```
Comment (ppt-adsk) [inline lib/mayaUsd/fileio/primUpdaterManager.cpp:1026]: Essentially, we are checking for a particular node type here --- and I don't think that's a good idea.  I would really hope we can implement an override of the discardEdits() virtual on MayaReferenceUpdater and do the work there.

**Review by ppt-adsk — CHANGES_REQUESTED** [review]
Any way we can test this automatically?

File: lib/mayaUsd/CMakeLists.txt (hunk 1 above)
Comment (pierrebai-adsk) [reply lib/mayaUsd/CMakeLists.txt:203]: It was necessary to access the schema attribute.

File: lib/usd/translators/mayaReferenceUpdater.cpp [hunk 3]
```diff
@@ -42,6 +44,7 @@
 #include <pxr/usd/usdUtils/pipeline.h>
 
 #include <maya/MFnAttribute.h>
+#include <ufe/pathString.h>
```
Comment (ppt-adsk) [inline lib/usd/translators/mayaReferenceUpdater.cpp:47]: Don't think we need this one.

**Review by ppt-adsk — APPROVED** [review]

**Pull request merged**